crates/services/p2p/src/p2p_service.rs

```diff
@@ -1036,6 +1041,31 @@ mod tests {
         stop_sender.send(()).unwrap();
     }
 
+    #[tokio::test]
+    #[instrument]
+    async fn our_node_in_reserved_nodes() {
+        let mut p2p_config = Config::default_initialized("own_node_in_reserved_nodes");
+        p2p_config.address = IpAddr::V4(Ipv4Addr::from([0, 0, 0, 0]));
+        p2p_config.tcp_port = 11111;
```

Comment:
When we run many tests, this port can be already used and then test will be incorrect. Maybe it is better to request the port from the OS and start a node on this port. It may require modification of the `build_service_from_config` to allow bypassing of the port binding.

Reply:
I found a solution that is IMO the best solution (that doesn't require to change all of the code just for this test) and as it seems a bit hacky I added a big comment that, I hope, explain all decisions taken.